homeassistant/components/media_player/soundtouch.py

```diff
@@ -0,0 +1,593 @@
+
+    def _set_media_album(self, source, doc):
+        if source in ['SPOTIFY', 'STORED_MUSIC']:
+            self._media_album_name =\
+                doc.getElementsByTagName("album")[0].firstChild.nodeValue
+        else:
+            self._media_album_name = None
+
+    def _set_media_artist(self, source, doc):
+        if source in ['SPOTIFY', 'STORED_MUSIC']:
+            self._media_artist =\
+                doc.getElementsByTagName("artist")[0].firstChild.nodeValue
+        else:
+            self._media_artist = None
+
+    def _set_media_track(self, source, doc):
+        if source in ['SPOTIFY', 'STORED_MUSIC']:
+            self._media_track = \
+                doc.getElementsByTagName("track")[0].firstChild.nodeValue
+        else:
+            self._media_track = None
+
+    def _set_media_title(self, source, doc):
+        if source in ['SPOTIFY', 'STORED_MUSIC']:
+            self._media_title = self.media_artist + " - " + self.media_track
+        elif source == 'INTERNET_RADIO':
+            self._media_title = doc.getElementsByTagName("stationName")[
+                0].firstChild.nodeValue
+
+    def _set_media_duration(self, source, doc):
+        if source in ['SPOTIFY', 'STORED_MUSIC']:
+            self._media_duration = \
+                int(doc.getElementsByTagName("time")[0].attributes[
+                    "total"].value)
+        else:
+            self._media_duration = None
+
+    def update(self):
+        """Retrieve the latest data."""
+        doc = self._get_status()
+        source = doc.getElementsByTagName("ContentItem")[0].attributes[
+            "source"].value
+        if source == 'STANDBY':
+            self._state = STATE_OFF
+            self._reset_properties()
+        elif source == 'INVALID_SOURCE':
+            self._state = STATE_UNAVAILABLE
+            self._reset_properties()
+        else:
+            status = doc.getElementsByTagName("playStatus")[
+                0].firstChild.nodeValue
+            if status == "PLAY_STATE":
+                self._state = STATE_PLAYING
+            elif status == "PAUSE_STATE":
+                self._state = STATE_PAUSED
+            else:
+                self._state = STATE_UNKNOWN
+
+            if self._state in [STATE_PLAYING, STATE_PAUSED]:
+                self._set_image_url(source, doc)
+                self._set_media_album(source, doc)
+                self._set_media_artist(source, doc)
+                self._set_media_track(source, doc)
+                self._set_media_title(source, doc)
+                self._set_media_duration(source, doc)
+
+    def _get_status(self):
+        response = requests.get(
+            'http://' + self._config['host'] + ":" + str(self._config['port'])
+            + '/now_playing')
+        doc = minidom.parseString(response.text)
+        return doc
+
+    def _send_key(self, key):
+        action = '/key'
+        press = '<key state="press" sender="Gabbo">%s</key>' % key
```

Comment:
Thanks for the feedback.

All this component contains specific protocol so it is a major refactoring.
I'll start working on it and create a dedicated library without link to Home Assistant.

It will maybe be easier to close this PR and I'll open a new one after. What do you think ?